src/evo/deterministicmns.cpp

```diff
@@ -214,16 +214,41 @@ CDeterministicMNCPtr CDeterministicMNList::GetMNPayee(const CBlockIndex* pIndex)
     return best;
 }
 
-std::vector<CDeterministicMNCPtr> CDeterministicMNList::GetProjectedMNPayeesAtChainTip(int nCount) const
-{
-    return GetProjectedMNPayees(::ChainActive()[nHeight], nCount);
-}
-
-std::vector<CDeterministicMNCPtr> CDeterministicMNList::GetProjectedMNPayees(const CBlockIndex* const pindex, int nCount) const
+std::vector<CDeterministicMNCPtr> CDeterministicMNList::GetProjectedMNPayees(int nCount) const
 {
     if (nCount < 0 ) {
         return {};
     }
+
+    bool isMNRewardReallocation{false};
+    if (auto pindex = ::ChainActive()[nHeight]; pindex == nullptr) {
```

Comment:
I don't like using here global variable ::ChainActive().Tip().
Also there can be 2 different chain (2 forks on same time) and extracting pindex by [nHeight] can be incorrect value. if after mnList is generated the chain would be switched to other one; the value of `::ChainActive()[nHeight]` would be not expected and list would be wrong.

For example call below:
```
auto projection = deterministicMNManager->GetListForBlock(pindexTip).GetProjectedMNPayees(20);
 ```
`pindexTip` is known but we lost this information.


btw, `GetProjecctedMNPayees` is just estimation list; is not used in consensus rules; so, I don't think that we need to consider case of LOCKED_IN to make calculation 100% exact for case that happens only once in chain - during MN_RR activation, I vote for simplicity here.

Reply:
also, if we will change exact known `pindex` by using `::ChainActive()[nHeight]` it may cause similar bug not only in GUI code but in all usages of `GetProjecctedMNPayees` (that are non-GUI only), so in this case it can cause this issue getting bigger rather than beeing solved and we won't see it anymore because no crash -> seems works (but it does not)